packages/devtools_app/lib/src/screens/debugger/codeview.dart

```diff
@@ -295,123 +311,133 @@ class _CodeViewState extends State<CodeView>
-                                      scrollController: textController,
-                                      lines: lines,
-                                      pausedFrame: pausedFrame,
-                                      searchMatchesNotifier:
-                                          widget.controller.searchMatches,
-                                      activeSearchMatchNotifier:
-                                          widget.controller.activeSearchMatch,
-                                    ),
+                                    debugController: widget.controller,
+                                    scrollController: textController,
+                                    lines: lines,
+                                    pausedFrame: pausedFrame,
+                                    searchMatchesNotifier:
+                                        widget.controller.searchMatches,
+                                    activeSearchMatchNotifier:
+                                        widget.controller.activeSearchMatch,
                                   ),
                                 ),
-                              );
-                            },
-                          ),
+                              ),
+                            );
+                          },
                         ),
-                      ],
-                    );
-                  },
-                ),
+                      ),
+                    ],
+                  );
+                },
               ),
             ),
-          );
-        } else {
-          return Expanded(
-            child: Center(
-              child: Text(
-                'No source available',
-                style: theme.textTheme.subtitle1,
-              ),
+          ),
+        );
+      } else {
+        return Expanded(
+          child: Center(
+            child: Text(
+              'No source available',
+              style: theme.textTheme.subtitle1,
             ),
-          );
-        }
-      },
+          ),
+        );
+      }
+    };
+    if (widget.enableHistory) {
+      return HistoryViewport(
+        history: widget.controller.scriptsHistory,
+        generateTitle: (ScriptRef? script) {
+          final scriptUri = script?.uri;
+          if (scriptUri == null) return '';
+          return scriptUri;
+        },
+        onTitleTap: () => widget.controller.toggleFileOpenerVisibility(true),
+        controls: [
+          ScriptPopupMenu(widget.controller),
+          ScriptHistoryPopupMenu(
+            itemBuilder: _buildScriptMenuFromHistory,
+            onSelected: (scriptRef) {
+              widget.controller.showScriptLocation(ScriptLocation(scriptRef));
+            },
+            enabled: widget.controller.scriptsHistory.hasScripts,
+          ),
+        ],
+        contentBuilder: contentBuilder,
+      );
+    }
+    // TODO(bkonyi): this is a hack to deal with some Expanded elements
```

Comment:
@kenzieschmoll is there an easier way to do this?

Reply:
I totally forgot I had this here. I still need to investigate this before this is ready to actually land.